File: 2211-count-collisions-on-a-road/2211-count-collisions-on-a-road.py
class Solution:
    def countCollisions(self, directions: str) -> int:
        res = 0
        st = []
        for i in directions:
            if i == 'R':
                st.append(i)
            elif i == 'L':
                if st:
                    if st.pop() == 'R':
                        res += 2
                        while st and st[-1] == 'R':
                            res += 1
                            st.pop()
                    else:
                        res += 1
                    st.append('S')
            else:
                while st and st[-1] == 'R':
                    res += 1
                    st.pop()
                st.append(i)
            # print(res)
        return res

File: 2211-count-collisions-on-a-road/2211-count-collisions-on-a-road.cpp
class Solution {
public:
    int countCollisions(string directions) {
        int n = directions.length();
        int lc = 0;
        int rc = n-1;
        int sc = 0;
        while (directions[lc] == 'L') {
            lc ++;
        }
        while (rc >= 0 && directions[rc] == 'R') {
            rc --;
        }
        for (int i = lc; i <= rc ; i++) {
            if (directions[i] != 'S') sc ++;
        }
        return sc;
    }
};